Update AltTester Version, Simplify Page Object Model and Improve avoidObstacles Method

SiminaRe opened the pull request:
### Summary of Changes
- **Updated AltTester version**: upgraded AltTester dependency to 2.1.2
  
- **Refactored code to align with Page Object Model**: reorganized the code structure to better adhere to the Page Object Model design pattern, to improve code readability and maintainability 

- **Improved avoidObstacles method**: enhanced the `avoidObstacles` method in `GamePlayPage` class so that the tests calling it are relevant and valid. 

Please review the changes and let me know if any further adjustments are needed.

SiminaRe committed:
refactor code to align with Page Object Model

- reorganized the code structure to better adhere to the Page Object Model design pattern, to improve code readability and maintainability
- removed AppTest ( AppTest class doesn't test any specific functionality of the application, it served as a very basic template for tests implementing JUnit 3.)
- fixed testShopPageCanBeClosed
- fixed multiple scene load in GamePlayTest
- update AltTester version
- added .idea in gitignore

SiminaRe committed:
removed AppTest

AppTest class doesn't test any specific functionality of the application, it served as a very basic template for tests implementing JUnit 3.

SiminaRe committed:
restructure of start page related code

SiminaRe committed:
restructure of main menu page related code

SiminaRe committed:
restructure of shop page related code

SiminaRe committed:
fixed testShopPageCanBeClosed

SiminaRe committed:
fixed multiple scene load

SiminaRe committed:
restructure of game play related code

SiminaRe committed:
game play tests updates

SiminaRe committed:
gameplay updates

SiminaRe committed:
correction in sorting obstacles formula

SiminaRe committed:
Game play debugging

SiminaRe committed:
added logging for avoidObstacles

SiminaRe committed:
update JUnit and AltTester versions; added Displayable interface

SiminaRe committed:
delete displayable interface

an interface wasn't needed, an abstract method in BasePage is enough

SiminaRe committed:
Added .idea in gitignore

SiminaRe committed:
revert update to junit5

revert update to junit5 so that allure reports are still generated. will rework junit update

SiminaRe committed:
clean code of logging statements

SiminaRe committed:
remove logging statements

rucindrea commented on src/main/java/pages/GamePlayPage.java:
```diff
@@ -12,132 +13,147 @@
+    @Override
     public boolean isDisplayed(){
-        if(pauseButton != null && character != null){
+        if(getPauseButton() != null && getCharacter() != null){
```
you could return `(getPauseButton() != null && getCharacter() != null)` directly

rucindrea commented on src/main/java/pages/MainMenuPage.java:
```diff
@@ -10,73 +10,76 @@
+    @Override
     public boolean isDisplayed(){
-        if(storeButton != null && leaderBoardButton != null && settingsButton != null && missionButton != null && runButton != null && characterName != null && themeName != null){
+        if(getStoreButton()!= null && getLeaderBoardButton() != null && getSettingsButton() != null && getMissionButton() != null && getRunButton() != null && getCharacterName() != null && getThemeName() != null){
```
same as above

rucindrea commented on src/main/java/pages/StartPage.java:
```diff
@@ -6,57 +6,49 @@
     }
+    @Override
+    public boolean isDisplayed() {
+        return (getStartButton() != null && getStartText() != null && getLogoImage() != null && getUnityUrlButton() != null);
```
nice :)

rucindrea reviewed (commented):
Overall, this looks good! I just added a few small comments.

SiminaRe committed:
Add test for checking the distance run by the cat

SiminaRe committed:
Add test for checking the distance run by the cat

SiminaRe committed:
collect fishes close

backup

SiminaRe committed:
add testCollectFishBonesOnMiddleLane

Add a test that compares the number of coins displayed at the end against the number of fishbones the cat collect from the middle lane. This is a "good-enough" test that covers a good amount of cases. It doesn't cover the cases where the cat reaches 150m+, when origin resets occurs for most fishbones; this needs to be handled

SiminaRe committed:
better results matching numbers

SiminaRe committed:
safe point

SiminaRe committed:
sometimes-works version

backup in care I mess the code badly again

SiminaRe committed:
test to check coins against collected middle lane fiishbones

Good enough test that covers a good amount of cases. It doesn't cover the cases where the cat reaches 150m+, in which origin resets occurs for most fishbones and which needs to be handled

SiminaRe committed:
attempt to fix fish collection

SiminaRe committed:
clean code for testCollectFishBonesOnMiddleLane and testDistanceRun

SiminaRe committed:
refactor idDisplayed methods

as suggested in code review

SiminaRe committed:
refactor idDisplayed methods

as suggested in code review

SiminaRe committed:
Merge branch 'master' of https://github.com/SiminaRe/EXAMPLES-Java-Standalone-and-Android-AltTrashCat
